build.cake

```diff
@@ -176,18 +178,31 @@ Task("SetupMSBuild")
 
     CopyDirectory(msbuildNetCoreAppInstallFolder, msbuildNetCoreAppFolder);
 
-    // Finally, copy Microsoft.CSharp.Core.targets from Microsoft.Net.Compilers
-    var csharpTargetsName = "Microsoft.CSharp.Core.targets";
-    var csharpTargetsPath = CombinePaths(env.Folders.Tools, "Microsoft.Net.Compilers", "tools", csharpTargetsName);
+    // Finally, copy Microsoft.Net.Compilers
+    var roslynFolder = CombinePaths(env.Folders.Tools, "Microsoft.Net.Compilers", "tools");
+    var roslynNet46Folder = CombinePaths(msbuildNet46Folder, "Roslyn");
+    var roslynNetCoreAppFolder = CombinePaths(msbuildNetCoreAppFolder, "Roslyn");
 
-    var csharpTargetsNet46Folder = CombinePaths(msbuildNet46Folder, "Roslyn");
-    var csharpTargetsNetCoreAppFolder = CombinePaths(msbuildNetCoreAppFolder, "Roslyn");
+    CreateDirectory(roslynNet46Folder);
+    CreateDirectory(roslynNetCoreAppFolder);
 
-    CreateDirectory(csharpTargetsNet46Folder);
-    CreateDirectory(csharpTargetsNetCoreAppFolder);
+    CopyDirectory(roslynFolder, roslynNet46Folder);
+    CopyDirectory(roslynFolder, roslynNetCoreAppFolder);
 
-    CopyFile(csharpTargetsPath, CombinePaths(csharpTargetsNet46Folder, csharpTargetsName));
-    CopyFile(csharpTargetsPath, CombinePaths(csharpTargetsNetCoreAppFolder,csharpTargetsName));
+    // Delete unnecessary files
+    foreach (var folder in new[] { roslynNet46Folder, roslynNetCoreAppFolder })
+    {
+        DeleteFile(CombinePaths(folder, "Microsoft.CodeAnalysis.VisualBasic.dll"));
+        DeleteFile(CombinePaths(folder, "Microsoft.VisualBasic.Core.targets"));
+        DeleteFile(CombinePaths(folder, "VBCSCompiler.exe"));
+        DeleteFile(CombinePaths(folder, "VBCSCompiler.exe.config"));
+        DeleteFile(CombinePaths(folder, "csi.exe"));
```

Comment:
you mean to have such APIs built into OmniSharp? yes that would be possible, I remember discussing this with Jason some time ago.
it could also be a third party extension, I built one like that [for scriptcs](https://marketplace.visualstudio.com/items?itemName=filipw.scriptcsRunner) long time ago

But the simple way would be to just have CSI.exe as part of the bundle. The main reason is there is no distribution channel for CSI.exe at the moment other than installing MSBuild tools on your machine (and no distribution channel for non-Windows platforms at all), which is quite annoying

Again, this should be discussed separately, as it doesn't impact this PR, but it would be great to be able to simply ship CSI.exe with OmniSharp so that as a user you don't need to install anything else if you i.e. just want to start a C# REPL or something like that. The C# extension in VS Code could just facilitate shelling scripts into CSI (again, to simulate a bit the experience you have in VS with C# Interactive Window) and give folks an exploratory playground for C# straight up.